testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py

```diff
@@ -3657,6 +3657,269 @@ def test_doc_removal_with_multipleChannels(params_from_base_test_setup, setup_cu
+        4. Do pull replication from SGW .
+        5. Verify docs got replicated to CBL dB from only one channel
+        6. Add 2nd role to the user .
+        8. Add new docs to SGW in both channels.
+        9.Continue to pull replication from SGW .
+        10.Verify all new docs got replicated from both channels
+
+    """
+    sg_db = "db"
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+    base_url = params_from_base_test_setup["base_url"]
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    db = params_from_base_test_setup["db"]
+    cbl_db = params_from_base_test_setup["source_db"]
+    sync_gateway_version = params_from_base_test_setup["sync_gateway_version"]
+
+    username = "autotest"
+    password = "password"
+    num_docs = 10
+
+    if sync_gateway_version < "2.5.0":
+        pytest.skip('This test cannnot run with sg version below 2.5.0')
+
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    role1_channels = ["ABC", "DEF"]
+    role2_channels = ["xyz", "XXX"]
+    role_name = "cache_role"
+    roles = [role_name]
+    new_role_name = "new_role"
+
+    sg_client = MobileRestClient()
+    authenticator = Authenticator(base_url)
+    replicator = Replication(base_url)
+
+    # 1. Create user.
+    sg_client.create_user(sg_admin_url, sg_db, username, password=password, roles=roles)
+    cookie, session_id = sg_client.create_session(sg_admin_url, sg_db, username)
+    session = cookie, session_id
+
+    # 2. Add new roles.
+    sg_client.create_role(url=sg_admin_url, db=sg_db, name=role_name, channels=role1_channels)
+    sg_client.create_role(url=sg_admin_url, db=sg_db, name=new_role_name, channels=role2_channels)
+
+    # 3. Create docs on SGW.
+    sg_client.add_docs(url=sg_admin_url, db=sg_db, number=num_docs, id_prefix="role_doc",
+                       channels=role1_channels, auth=session)
+
+    # 4. Do pull replication from SGW.
+    replicator_authenticator = authenticator.authentication(session_id, cookie, authentication_type="session")
+    repl = replicator.configure_and_replicate(source_db=cbl_db,
+                                              target_url=sg_blip_url,
+                                              continuous=True,
+                                              replication_type="pull",
+                                              replicator_authenticator=replicator_authenticator)
+    replicator.wait_until_replicator_idle(repl)
+
+    # 5. verify docs got replicated to CBL
+    cbl_doc_ids = db.getDocIds(cbl_db)
+    cbl_docs = db.getDocuments(cbl_db, cbl_doc_ids)
+    assert len(cbl_docs) == num_docs, "Docs did not get replicated to CBL"
+
+    # 6. Add 2nd role to the user .
+    sg_client.update_user(url=sg_admin_url, db=sg_db, name=username, roles=[role_name, new_role_name])
```

Comment:
to make code more readable, here new_role_name may be added to roles param and make this assignment as roles=roles as create_user did. or remove roles = [role_name], and directly passing [role_name] to create_user. anyway this is minor and my personal opinion.